R/pkg/R/DataFrame.R

```diff
@@ -2624,6 +2624,15 @@ setMethod("except",
 setMethod("write.df",
           signature(df = "SparkDataFrame"),
           function(df, path = NULL, source = NULL, mode = "error", ...) {
+            if (!is.character(path) && !is.null(path)) {
+              stop("path should be charactor, null or omitted.")
```

Comment:
minor point: is it more efficient to flip the checks, ie.
`if (!is.null(path) && !is.character(path))`, since path defaults to NULL?